src/test/java/org/inferred/freebuilder/processor/ListSourceTest.java

```diff
@@ -437,14 +426,298 @@ public void test_guava_j7() {
+        "    private final List<Integer> age;",
+        "",
+        "    private Value(Person_Builder builder) {",
+        "      this.name = ImmutableList.copyOf(builder.name);",
+        "      this.age = ImmutableList.copyOf(builder.age);",
+        "    }",
+        "",
+        "    @Override",
+        "    public List<String> getName() {",
+        "      return name;",
+        "    }",
+        "",
+        "    @Override",
+        "    public List<Integer> getAge() {",
+        "      return age;",
+        "    }",
+        "",
+        "    @Override",
+        "    public boolean equals(Object obj) {",
+        "      if (!(obj instanceof Person_Builder.Value)) {",
+        "        return false;",
+        "      }",
+        "      Person_Builder.Value other = (Person_Builder.Value) obj;",
+        "      return Objects.equals(name, other.name) && Objects.equals(age, other.age);",
+        "    }",
+        "",
+        "    @Override",
+        "    public int hashCode() {",
+        "      return Objects.hash(name, age);",
+        "    }",
+        "",
+        "    @Override",
+        "    public String toString() {",
+        "      return \"Person{\" + \"name=\" + name + \", \" + \"age=\" + age + \"}\";",
+        "    }",
+        "  }",
+        "",
+        "  private static final class Partial extends Person {",
+        "    private final List<String> name;",
+        "    private final List<Integer> age;",
+        "",
+        "    Partial(Person_Builder builder) {",
+        "      this.name = ImmutableList.copyOf(builder.name);",
+        "      this.age = ImmutableList.copyOf(builder.age);",
+        "    }",
+        "",
+        "    @Override",
+        "    public List<String> getName() {",
+        "      return name;",
+        "    }",
+        "",
+        "    @Override",
+        "    public List<Integer> getAge() {",
+        "      return age;",
+        "    }",
+        "",
+        "    @Override",
+        "    public boolean equals(Object obj) {",
+        "      if (!(obj instanceof Person_Builder.Partial)) {",
+        "        return false;",
+        "      }",
+        "      Person_Builder.Partial other = (Person_Builder.Partial) obj;",
+        "      return Objects.equals(name, other.name) && Objects.equals(age, other.age);",
+        "    }",
+        "",
+        "    @Override",
+        "    public int hashCode() {",
+        "      return Objects.hash(name, age);",
+        "    }",
+        "",
+        "    @Override",
+        "    public String toString() {",
+        "      return \"partial Person{\" + COMMA_JOINER.join(\"name=\" + name, \"age=\" + age) "
+            + "+ \"}\";",
+        "    }",
+        "  }",
+        "}\n"));
+  }
+
+  @Test
+  public void test_guava_j8() {
+    Metadata metadata = createMetadata();
+
+    String source = generateSource(
+        metadata,
+        JAVA_7,  // Currently also represents Java 8
```

Comment:
given that you're basically never going to change this behaviour (as it'll be a breaking change for older compilers etc), it might be worth calling this JAVA_7_OR_8 or something.